src/simulator/vacuum_cleaner.py

```diff
@@ -1,51 +1,53 @@
 from math import dist
-from turtle import position
 from object import ObjectId, Object
 from robot import Robot
 from position import Position
 from cylinder import Cylinder
 from sensor import Sensor
 from bumper_sensor import BumperSensor
+from object_registry import ObjectRegistry
 import numpy as np
+import copy
 
 
 class VacuumCleanerV0(Robot):
-    def __init__(self, oid: ObjectId, position: Position, owner_object:Object, \
-        parameters: dict[str,]):
+    def __init__(self, oid: ObjectId, name: str, position: Position, owner_object: Object, \
+                 parameters: dict[str,], registry: ObjectRegistry):
         # TODO saeed: making assumption that from the usual inputs of Object constructor
         #   we only need position. e.g. robots have no owner? it still doesn't hurt so I
         #   include it.
         # TODO saeed: class-associated object, needs parameters. How do we collect these from
         #   map and pass?
-        
-        #assuming Robot didn't override Object's constructor
-        Robot.__init__(self,oid,\
-            Cylinder(parameters["diameter"],parameters["height"]),position, owner_object)
-        
+        # assuming Robot didn't override Object's constructor
+        Robot.__init__(self, oid, name, \
+                       Cylinder(parameters["diameter"], parameters["height"]), position, owner_object, registry)
         # TODO saeed: where does the oid of sensor comes from? fetched from Map? how do we have
         #  acess to Map here?
+        self.sensor_id = self.registry.get_next_available_id()
```

Comment:
Should vacuum cleaner stores its sensor's id as a member variable? To me it seems that each object should encapsulate its own data. VC can use `self.sensor.id` to access the id (or better, `self.sensor.getID()`, but I just realized we don't have such a method yet).

Reply:
You're right. I think `Object` has to have the method` Object.getID()` so that all of its children inherit this method. Should I make this change now and adjust the whole code according to it?
By the way, your comment popped up a question into my head: Outside of `Object` and its children's classes, one should not be able to change the id of an object, so why `id` is not a protected property?